src/modules/SX126x.cpp

```diff
@@ -1058,8 +1063,25 @@ int16_t SX126x::setFrequencyRaw(float freq) {
   return(ERR_NONE);
 }
 
+int16_t SX126x::setDio2AsRfSwitch(bool enable) {
+  uint8_t data[1];
+  if (enable) {
+  // set DIO2 as RF switch
+    data[0] = SX126X_DIO2_AS_RF_SWITCH;
+  } else {
+    data[0] = SX126X_DIO2_AS_IRQ;
+  }
+  int16_t state = SPIwriteCommand(SX126X_CMD_SET_DIO2_AS_RF_SWITCH_CTRL, data, 1);
+
+  if (state == ERR_NONE) {
+    _dio2RfSwitch = true;
+  }
+  return(state);
+}
+
 int16_t SX126x::config(uint8_t modem) {
   // set DIO2 as IRQ
```

Comment:
It would be better to remove this entire block. The purpose of config() metod is to configure stuff that is inaccessible via the public API. DIO2 setting is accessible now, so we should have a call `setDio2AsRfSwitch(false);` in the `begin()` and `beginFSK()` mtehods.